### PR TITLE
sn - Create CommonsTable, and CommonsTest, Add tests to Table and Util, and fix coverage

### LINKED ISSUE #35: 6pm-4: Make CommonsTable and commonsUtil
add the CommonsTable component to the project.

- [x] The table has columns for database fields (i.e. the attributes of the corresponding `@Entity` class
- [x] There are fixtures under `/frontend/src/fixtures` for this data type, similar to the other fixtures files.
- [x] There is a storybook entry similar to the one for `UCSBDatesTable`
- [x] 100% coverage with `npm run coverage`
- [x] 100% coverage with `npx stryker run`

### LINKED ISSUE #48: 6pm-4: add tests for Table and Util
- [x] `CommonsTable.js` has 100% test coverage
- [x] `CommonsTable.js` has full mutation coverage
- [x] `commonsUtil.js` has 100% test coverage
- [x] `commonsUtil.js` has full mutation coverage

### PR DESCRIPTION
**Pull request by sahilnaik1**
# Overview

This PR creates CommonsTable, commonsUtil, and tests for commonsUtil.js and CommonsTable.js Coverage fixed and now at 100%. fixtures rebased

# Issues Addressed

If this PR closes or deals with issues, mention them and tag the issue number with the `#` syntax.

# Details

Where applicable, include "before" and "after" screenshots or animated GIFs.   The tool [LiceCAP](https://www.cockos.com/licecap/) or one of [these similar tools](https://www.nextofwindows.com/5-free-tools-to-screen-capture-to-gif-on-windows) 
can be used to capture screenshots/animated GIFs.

## Reminders (delete this section before merging or as code is merged)
- [ ] Explicitly link to this pull request each issue that it closes
- [ ] Assign one and ideally two reviewers to review this pull request. 
- [ ] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

### CODE REVIEW

File: frontend/src/main/components/Commons/CommonsTable.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            accesor:'totalPlayers',
+        },
+        {
+            Header:'cowPrice',
+            accesor:'cowPrice',
+        },
+        {
+            Header:'milkPrice',
+            accesor:'milkPrice',
+        },
```
Comment (btk5h) [inline frontend/src/main/components/Commons/CommonsTable.js]: Typos? Should be `accessor` not `accesor`.

File: frontend/src/tests/utils/commonsUtil.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    //         // act
+    //         const result = editCallback(cell);
+
+    //         // assert
+
+    //         expect(mockToast).toHaveBeenCalledWith(expectedToast);
+           
+    //     });
+
+    // });
```
Comment (btk5h) [inline frontend/src/tests/utils/commonsUtil.test.js]: Can you remove this commented code?

File: frontend/src/main/pages/AdminListCommonPage.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+import CommonsTable from 'main/components/Commons/CommonsTable';
+import { useBackend } from 'main/utils/useBackend';
+//import { useCurrentUser } from 'main/utils/currentUser'
```
Comment (btk5h) [inline frontend/src/main/pages/AdminListCommonPage.js]: Can you remove this commented import?

File: frontend/src/main/components/Commons/CommonsTable.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    ];
+
+    if (hasRole(currentUser, "ROLE_ADMIN")) {
+        columns.push(ButtonColumn("Edit", "primary", editCallback, "CommonsTable"));
+        columns.push(ButtonColumn("Delete", "danger", deleteCallback, "CommonsTable"));
+    } 
+
+    // Stryker disable next-line ArrayDeclaration : [columns] is a performance optimization
+    const memoizedColumns = React.useMemo(() => columns, [columns]);
+    const memoizedCommons = React.useMemo(() => commons, [commons]);
```
Comment (btk5h) [inline frontend/src/main/components/Commons/CommonsTable.js]: This does not actually memoize these fields. In order for this to actually be a performance optimization, the function within `useMemo` must generate the array. As it is currently, this actually negatively impacts performance.

File: frontend/src/tests/utils/commonsUtil.test.js (hunk 2 above)
Comment (btk5h) [inline frontend/src/tests/utils/commonsUtil.test.js]: Can you remove this commented code?

**Review by btk5h — DISMISSED** [review]

**Review by rsdharmaji — APPROVED** [review]

**Comment by jacksoncooper** [comment]
Closing, merging to #40.

**Pull request closed without merging**